test/Parse/recovery.swift

```diff
@@ -624,6 +624,8 @@ struct InitializerWithName {
 
 struct InitializerWithNameAndParam {
   init a(b: Int) {} // expected-error {{initializers cannot have a name}} {{8-9=}}
+  init y?(_ y: Int) {} // expected-error {{initializers cannot have a name}} {{8-9=}}
+  init z<T>(y: T) {} // expected-error {{initializers cannot have a name}} {{8-9=}}
```

Comment:
I think I agree about `init z<T>`, but `init? y` seems more likely than `init y?` to me. Not that it really matters, this is all recovery.

Reply:
Agreed.